### DISCUSSION TITLE
Does rtdl support multiple tokens?

### DISCUSSION THREAD

**Discussion by Up-in-the-wind** [yandex-research/rtdl#60] [Q&A]
Dear sir/madam：
There is a certain relationship between my different pieces of data. I would like to use each piece of data (array shape) as a token and input multiple tokens into FT-Transformer at the same time. How can I achieve this?

**Comment by Yura52** [comment]
Hi! Do I understand correctly that you have some data that is already embedded, and you don't need the "Feature Tokenizer" part ("FT-" in "FT-Transformer")? If yes, then, perhaps, [rtdl.Transformer](https://github.com/Yura52/rtdl/blob/b354b35d68f28b4f5bbebd2e6d5b1f6cfa91eed1/rtdl/modules.py#L921) is what you are looking for? In [its forward method](https://github.com/Yura52/rtdl/blob/b354b35d68f28b4f5bbebd2e6d5b1f6cfa91eed1/rtdl/modules.py#L1139), it accepts a generic tensor of the shape `(batch_size, n_tokens, d_embedding)`. `rtdl.FTTransformer` uses `rtdl.Transformer` under the hood.

P.S. I hope that in the upcoming two weeks I will release a refactored version of FT-Transformer, perhaps, it will be easier to experiment with. With that release, `rtdl.FTTransformer` and `rtdl.Transformer` will be deprecated.

**Comment by Up-in-the-wind** [comment]
Hi, thanks you for the quick feedback and  providing the solution ideas. I am also very looking forward to the update of rtdl.
